pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// nolint:typecheck,goconst,gocritic,stylecheck,nolintlint
+package shared
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/constants"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	kappctrl "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	kapppkgiv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+	addonutil "github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
+	runtanzuv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+)
+
+const (
+	waitTimeout     = time.Second * 90
+	pollingInterval = time.Second * 2
+)
+
+// create cluster client from kubeconfig
+func createClientFromKubeconfig(exportFile string, scheme *runtime.Scheme) (client.Client, error) {
+	config, err := clientcmd.LoadFromFile(exportFile)
+	Expect(err).ToNot(HaveOccurred(), "Failed to load cluster Kubeconfig file from %q", exportFile)
+
+	rawConfig, err := clientcmd.Write(*config)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create raw config ")
+
+	restConfig, err := clientcmd.RESTConfigFromKubeConfig(rawConfig)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create rest config ")
+
+	client, err := client.New(restConfig, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred(), "Failed to create a cluster client")
+
+	return client, nil
+}
+
+// getClusterBootstrap gets ClusterBootstrap resource with the provided object key
```

Comment:
"provided object key" ?  I think you mean with provided clustername and namespace. no key is provided as a paramenter.  Also probably don't even need comment. Is a private function and is pretty obvious from just the function signature what it does.